app/Components/Sudoku Board/SudokuBoard.tsx

```diff
@@ -284,68 +289,85 @@ const SudokuBoard = (props: SudokuBoardProps) => {
   };
 
   const renderCell = (cell: CellProps, r: number, c: number) => {
-    // row and column values are incorrect here.
-    // console.log("RENDER CELL", c, "COLUMN", r, "ROW", cell)
-    const {
-      isHighlightIdenticalValues,
-      isHighlightBox,
-      isHighlightRow,
-      isHighlightColumn,
-    } = React.useContext(PreferencesContext);
-
-    let selected = sudokuBoard.selectedCell;
-    let isSelected = false;
-    let conflict = isConflict(r, c, cell);
-    let peer = false;
-    let box = false;
-    let row = false;
-    let column = false;
-    let sameValue = false;
-    if (selected != null) {
-      isSelected = c === selected.c && r === selected.r;
-      box = isCurrentCellAndSelectedCellInSameBox({ r: r, c: c }, selected);
-      row = isCurrentCellAndSelectedCellInSameRow({ r: r, c: c }, selected);
-      column = isCurrentCellAndSelectedCellInSameColumn(
-        { r: r, c: c },
-        selected
-      );
-      peer =
-        !conflict &&
-        ((box && isHighlightBox) ||
-          (row && isHighlightRow) ||
-          (column && isHighlightColumn));
-
-      const currentSelectedCell: CellProps | null = getCurrentSelectedCell();
-      if (currentSelectedCell == null) {
-        return;
-      }
-      const selectedEntry = currentSelectedCell.entry;
-      let currentEntry = cell.entry;
-      sameValue =
-        !conflict &&
-        isHighlightIdenticalValues &&
-        selectedEntry === currentEntry &&
-        currentEntry != 0;
+    const selectedCell = sudokuBoard.selectedCell;
+    const isSelected =
+      selectedCell == null
+        ? false
+        : c === selectedCell.c && r === selectedCell.r;
+    const conflict = isConflict(r, c, cell);
```

Comment:
Would be useful to have a multi-line comment above here to explain why conflicts, peers, and same values affect the background color. Maybe give examples?

Reply:
I feel like this would be better suited for our documentation. I added more documentation for Board Highlighting to go with the diagram you made.